scheduler/src/cook/mesos/schema.clj

```diff
@@ -945,7 +945,17 @@ for a job. E.g. {:resources {:cpus 4 :mem 3} :constraints {\"unique_host_constra
                          prior-state (:job/state job)]
                      (if (= prior-state :job.state/completed)
                        [[:db/add j :job/state :job.state/waiting]]
-                       []))}}])
+                       []))}}
+   {:db/id (d/tempid :db.part/user)
+    :db/ident :ensure-not-exists
+    :db/doc "Ensure that the given entity does not exist"
+    :db/fn #db/fn {:lang "clojure"
+                   :params [db field id]
```

Comment:
Change this to take `db` and `id-or-lookup-ref` and then call `(d/entity db id-or-lookup-ref)` that way it is more generic.